lightning/src/ln/msgs.rs

```diff
@@ -1550,6 +1550,13 @@ pub trait ChannelMessageHandler : MessageSendEventsProvider {
 	/// If it's `None`, then no particular network chain hash compatibility will be enforced when
 	/// connecting to peers.
 	fn get_chain_hashes(&self) -> Option<Vec<ChainHash>>;
+
+	/// Retries the [`InvoiceRequest`] message for pending outbound payments
+	/// that are still awaiting a [`Bolt12Invoice`].
+	///
+	/// [`InvoiceRequest`]: crate::offers::invoice_request::InvoiceRequest
+	/// [`Bolt12Invoice`]: crate::offers::invoice::Bolt12Invoice
+	fn retry_invoice_requests(&self) -> Result<(), ()>;
```

Comment:
Done! [2566089](https://github.com/lightningdevkit/rust-lightning/commit/2566089ec0b127e55febc934c43677a3d00d1716)

Rename the function to `handle_message_received` so it's name can be more generalized.
And move the function's call to `do_handle_message_without_peer_lock`, so that we don't need to update other functions explicitly.

Thanks for the pointer!